In [4]:
import pandas as pd
import numpy as np

df = pd.read_csv('benchmark_results.csv')
df.head()

query  chunk_size  chunk_overlap         embedding_function  chunk_count  \
0      0         500              0  multi-qa-MiniLM-L6-cos-v1            3   
1      1         500              0  multi-qa-MiniLM-L6-cos-v1            3   
2      2         500              0  multi-qa-MiniLM-L6-cos-v1            3   
3      3         500              0  multi-qa-MiniLM-L6-cos-v1            3   
4      4         500              0  multi-qa-MiniLM-L6-cos-v1            3   

   distance_max  distance_avr  input_tokens  rating  
0      0.568730      0.528078           411       3  
1      0.752624      0.631236           431      10  
2      0.759627      0.738991           366      10  
3      0.651457      0.565125           405       0  
4      0.706015      0.690337           416       6

In [5]:
df.groupby(['embedding_function'])[['distance_max', 'distance_avr', 'rating']].mean().reset_index()

embedding_function  distance_max  distance_avr    rating
0           all-MiniLM-L6-v2      0.734516      0.678089  7.411111
1  models/text-embedding-004      0.334607      0.309853  7.666667
2  multi-qa-MiniLM-L6-cos-v1      0.741642      0.683292  8.144444

In [33]:
df['distance_max'].quantile(0.90)

np.float64(0.9960561990737915)

In [23]:
df = df[df['embedding_function'] == 'multi-qa-MiniLM-L6-cos-v1']
df.count()

query                 180
chunk_size            180
chunk_overlap         180
embedding_function    180
chunk_count           180
distance_max          180
distance_avr          180
input_tokens          180
rating                180
dtype: int64

In [24]:
df.groupby(['chunk_size', 'chunk_overlap'])[['input_tokens','rating']].mean().sort_values(by='rating').reset_index()

chunk_size  chunk_overlap  input_tokens  rating
0         500              0        499.55    7.15
1         500            200        521.00    7.30
2         500            100        520.70    7.55
3        1000            100       1045.45    7.75
4        1000            200       1041.30    8.10
5        1500            200       1478.90    8.50
6        1000              0       1012.90    8.80
7        1500            100       1425.25    8.85
8        1500              0       1450.35    9.30

There is a more than point 1 (8.80 - 7.55) difference in rating between the configuration (chunk_size=500, chunk_overlap=100) and (chunk_size=1000, chunk_overlap=0).

In [25]:
df.groupby(['chunk_count'])[['input_tokens','rating']].mean().reset_index()

chunk_count  input_tokens    rating
0            3    786.300000  7.600000
1            5   1212.677778  8.688889

In [26]:
itokens_percentage = (1212-786)/786
rating_percentage = (8.68-7.6)/7.6

print(f"Input tokens percentage: {itokens_percentage:.2%}")
print(f"Rating percentage: {rating_percentage:.2%}")

Input tokens percentage: 54.20%
Rating percentage: 14.21%
